onnx/defs/experiments/defs.cc

```diff
@@ -106,28 +106,33 @@ NOTE: Currently, it supports data type of float, int32, int64, and bool.
     .Attr(
         "value",
         "The value for the elements of the output tensor.",
-        AttributeProto::FLOAT)
+        AttributeProto::FLOAT,
+        0.0f)
     .Attr(
         "dtype",
         "The data type for the elements of the output tensor."
         "Strictly must be one of the types from DataType enum in TensorProto.",
-        AttributeProto::INT)
+        AttributeProto::INT,
+        static_cast<int64_t>(TensorProto::FLOAT))
```

Comment:
Does the compiler complain if you remove the cast here? It would be convenient to not need to explicitly cast in all the schemas.

Reply:
Good catch!!! It was (there're another API name,desc,type, similar with name,type,int before. Now it should not complain. Let me fix it. I also hate it :).

Reply:
I would argue that in these schema declarations, readability is very important. (We can (and should) add tests for checking the cast if you have concern about the safely here).

Reply:
Not get the point here. It's not because I have concern and I add cast here. It's because compiler can't decide which API it should call/use without explicit type setting here.